lib/puppet/network/http/connection.rb

```diff
@@ -189,6 +190,42 @@ def request_with_redirects(request, options)
             request.each do |header, value|
               current_request[header] = value
             end
+          when 429, 503
+            retry_after = current_response['Retry-After']
+
+            if retry_after.nil?
+              # Set return value and advance to next loop.
+              response = current_response
+              next
+            end
+
+            retry_after = begin
```

Comment:
Minor nit - pull this out into a method rather than an inline anonymous for readability?

Reply:
Won't be straight-forward as the logic for terminating the outer retry loop is part of the error handling for this block.